code/modules/clothing/head/misc.dm

```diff
@@ -76,6 +76,7 @@
 /obj/item/clothing/head/bandana/green
 	name = "green bandana"
 	desc = "It's a green bandana with some fine nanotech lining."
+	flags_inv = BLOCKHAIR
```

Comment:
No need for that, /obj/item/clothing/head/bandana is the parent of this object. It already inherits it.

Reply:
If  I erase them, the hair doesn't dissapear in the others bandanas

Reply:
It won't, you just assigned something in flags_inv in /obj/item/clothing/head/bandana and it gets shared with childs as long they don't get modified again.